Java-Introduction.md

```diff
@@ -0,0 +1,123 @@
+
+   Open JDK installation :
+   ```bash
+   sudo apt-get install openjdk-8-jdk
+   ```
+
+   Oracle JDK installation :
+   ```bash
+   sudo add-apt-repository ppa:webupd8team/java
+   sudo apt-get update
+   sudo apt-get install oracle-java8-installer
+   ```
+
+### Mac
+
+* Either download Mac OSX .dmg executable from Oracle Downloads
+* Or use [Homebrew](http://brew.sh/) to [install](http://stackoverflow.com/a/28635465/2861269) :
+
+   ```bash
+   brew tap caskroom/cask
+   brew install brew-cask
+   brew cask install java
+   ```
+
+### Verify Installation
+
+Verify Java has been properly installed in your system by opening Command Prompt / Terminal and checking the versions of Java runtime and compiler :
+
+```bash
+$ java -version
+java version "1.8.0_66"
+Java(TM) SE Runtime Environment (build 1.8.0_66-b17)
+Java HotSpot(TM) 64-Bit Server VM (build 25.66-b17, mixed mode)
+
+$ javac -version
+javac 1.8.0_66
+```
+
+## JVM
+
+Ok now since we are done with the installations, lets begin to understand first the nitty gritty of the Java ecosystem. Java is an interpreted language, that is the code we write gets compiled and then run. We write the code in .java files, Java compiles them into [bytecodes](https://en.wikipedia.org/wiki/Java_bytecode) which are run on a Java Virtual Machine or JVM for execution. These bytecodes typically has a .class extension.
+
+Java is a pretty secured language as it doesn't let your program run directly on the machine. Instead, your program runs on a Virtual Machine called JVM. This Virtual Machine exposes several API's for low level machine interactions you can make, but other than that you cannot play with machine instructions explicitely. This adds a huge bonus of security.
+
+Also, once your bytecode is compiled it can run on any Java VM. This Virtual Machine is machine dependent, i.e it has different implementations for Windows, Linux and Mac. But your program is guranteed to run in any system thanks to this VM. This philosophy is called ["Write Once, Run Anywhere"](https://en.wikipedia.org/wiki/Write_once,_run_anywhere).
+
+## Hello World!
+
+Lets write a sample Hello World application. Open any editor / IDE of choice and create a file `HelloWorld.java`.
+
+```java
+public class HelloWorld {
+
+    public static void main(String[] args) {
+        // Prints "Hello, World" to the terminal window.
+        System.out.println("Hello, World");
+    }
+
+}
+```
+
+Now open the terminal / Command Prompt, and compile the file :
+
+```bash
+$ javac HelloWorld.java
+```
+
+Now run the file using `java` command!
+
+```bash
+$ java HelloWorld
+Hello, World
+```
+
+Congrats! Your first Java program has run successfully. Here we are just printing a string passing it to the API `System.out.println`. We will cover all the concepts in the code, but you are welcome to take a [closer look](https://docs.oracle.com/javase/tutorial/getStarted/application/)!
```

Comment:
If anything goes wrong here, the user might want some help. Link to gitter Java chatroom so that they clearly know where to ask for help.